unity/UnityEmbedHost.Tests/BaseEmbeddingApiTests.cs

```diff
@@ -616,6 +616,14 @@ static void AssertMultiDimensionalArraysAreEquivalent(Array expected, Array actu
         Assert.That(actualFlat, Is.EquivalentTo(expectedFlat));
     }
 
+    [TestCase(typeof(Animal),                    false)]
+    [TestCase(typeof(GenericAnimal<int, string>), true)]
```

Comment:
A few more test cases:
1. A generic type definition, `List<>`
2. Array of generics
3. pointer and by reference type if possible.